proposed/2022/PrivateAssetIndepdentFromExcludeAsset.md

```diff
@@ -0,0 +1,321 @@
+We shouldn't break customers who rely on current behavior, that is why're introducing new boolean `ExcludedAssetsFlow` metadata, which lets `PrivateAssets` fully control transitive flow of assets independently of whether they are included in the current project.
+
+In addition to the above code author should be able to decide which asset flow to parent consuming projects when this project is consumed as project reference(P2Ps that reference this project).
+If the current project is consuming any packages, then it can be transitively consumed by parent project if code author wants to.
+
+## Explanation
+
+### Functional explanation
+
+<!-- Explain the proposal as if it were already implemented and you're teaching it to another person. -->
+<!-- Introduce new concepts, functional designs with real life examples, and low-fidelity mockups or  pseudocode to show how this proposal would look. -->
+The new `ExcludedAssetsFlow` boolean metadata complements the `PrivateAssets` metadata to exclusively decide which assets flow to consuming parent project, but doesn't affect restore experience for current project so there would be no change in `project.assets.json` lock file, i.e it doesn't change `IncludeAssets/ExcludeAssets` calculation for current project.
+
+It'll change how `compile, runtime, contentFiles, build, buildMultitargeting, buildTransitive, analyzers, native` dependencies flow into the projects consuming it via `PackageReference` and `ProjectReference` references.
+
+For the following table assume `ExcludedAssetsFlow` opt-in metadata is set `true` in a package reference, iterating possible scenarios (not full list) for consuming parent project.
+
+| Asset flowing to parent project | New feature enabled | Possible downside |
+|-----------------------|--------------|-----------------|
+| compile | Expose new code, auto complete, intellicode, intellisense | Compile error due to naming ambiguity, run time error |
+| runtime | Let provide new runtime dependency | Run time error |
+| build | Enable msbuild imports | Build fails due to property/target value change |
+| analyzers | Code analyzers work | n/a |
+
+Recall our table from earlier that shows which assets flow based on listing some asset in `IncludeAssets` and/or `PrivateAssets`. Below we add a 4th column that reveals the new flow behavior when the new behavior is activated, notice how transitive flow is controlled exclusively by the `PrivateAssets` value when `ExcludedAssetsFlow` is set to `true` in column3 and row 3:
+
+IncludeAssets|PrivateAssets|Flows transitively (current)|Flows transitively (ExcludedAssetsFlow=true)
```

Comment:
The context from the previous table is low here, as now the reader has to look up what asset group is being talked about here.